compiler/rustc_next_trait_solver/src/canonicalizer.rs

```diff
@@ -183,46 +176,35 @@ impl<'a, 'tcx> Canonicalizer<'a, 'tcx> {
         for var in var_infos.iter_mut() {
             if var.is_region() {
                 assert!(var.is_existential());
-                *var = var.with_updated_universe(curr_compressed_uv);
+                let compressed_var = var.with_updated_universe(curr_compressed_uv);
```

Comment:
previously relied on `var` being `Copy`? I think we should keep the previous impl and require it to be `Copy` in the `Interner` trait if possible